core/sub.mk

```diff
@@ -26,3 +26,18 @@ $(foreach f, $(EARLY_TA_PATHS), $(eval $(call process_early_ta,$(f))))
 $(foreach f, $(CFG_IN_TREE_EARLY_TAS), $(eval $(call \
 	process_early_ta,$(out-dir)/ta/$(f).stripped.elf)))
 endif
+
+ifeq ($(CFG_EMBED_DTB),y)
+core-embed-fdt-dts = $(arch-dir)/dts/$(CFG_EMBED_DTB_SOURCE_FILE)
+core-embed-fdt-dtb = $(out-dir)/$(basename $(core-embed-fdt-dts)).dtb
+core-embed-fdt-c = $(out-dir)/$(arch-dir)/dts/embedded_secure_dtb.c
+gensrcs-y += embedded_secure_dtb
+cleanfiles += $(core-embed-fdt-c)
+produce-embedded_secure_dtb = arch/$(ARCH)/dts/embedded_secure_dtb.c
+depends-embedded_secure_dtb = $(core-embed-fdt-dtb) scripts/ta_bin_to_c.py
+recipe-embedded_secure_dtb = scripts/bin_to_c.py \
+				--bin $(core-embed-fdt-dtb) \
+				--vname embedded_secure_dtb \
+				--out $(core-embed-fdt-c)
+$(eval $(call gen-dtb-file,$(core-embed-fdt-dts),$(core-embed-fdt-dtb)))
+endif
```

Comment:
The generated C file should have the base name of the source DTS.
```make
ifeq ($(CFG_EMBED_DTB),y)
core-embed-fdt-dts = $(arch-dir)/dts/$(CFG_EMBED_DTB_SOURCE_FILE)
core-embed-fdt-dtb = $(out-dir)/$(arch-dir)/dts/$(CFG_EMBED_DTB_SOURCE_FILE:.dts=.dtb)
core-embed-fdt-c = $(out-dir)/$(arch-dir)/dts/$(CFG_EMBED_DTB_SOURCE_FILE:.dts=.c)
gensrcs-y += embedded_secure_dtb
cleanfiles += $(core-embed-fdt-c)
produce-embedded_secure_dtb = arch/$(ARCH)/dts/$(CFG_EMBED_DTB_SOURCE_FILE:.dts=.c)
depends-embedded_secure_dtb = $(core-embed-fdt-dtb) scripts/ta_bin_to_c.py
recipe-embedded_secure_dtb = scripts/bin_to_c.py \
				--bin $(core-embed-fdt-dtb) \
				--vname embedded_secure_dtb \
				--out $(core-embed-fdt-c)
$(eval $(call gen-dtb-file,$(core-embed-fdt-dts),$(core-embed-fdt-dtb)))
endif
```

Reply:
Ok, i move to your code.